### PR TITLE
Cleanup FairRadGridManager and FairMCApplication

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
* FairRadGridManager: Drop unused private member variables, better handling of singleton variable in destructor
* FairMCApplication: Drop unused private Operator
* FairMCApplication: Use unique_ptr for fRadGridMan

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Review by fuhlig1 — DISMISSED** [review]

**Comment by ChristianTackeGSI** [comment]
rebased, ready for review.

**Review by karabowi — APPROVED** [review]
I have tried your change with Geant4 MT, I wanted to check if the deletion of FairMCApplication's operator= causes any problem. It does not.
Tested with PandaRoot as well.

File: base/sim/FairMCApplication.h [hunk 1]
```diff
@@ -302,13 +303,15 @@ class FairMCApplication : public TVirtualMCApplication
     /** Current state */
     FairMCApplicationState fState;   //!
 
-    ClassDef(FairMCApplication, 4);
+    ClassDefOverride(FairMCApplication, 4);
```
Comment (fuhlig1) [inline base/sim/FairMCApplication.h:306]: Since you changed a data member, you should increase the version counter.

**Review by fuhlig1 — CHANGES_REQUESTED** [review]
See my change request.

File: base/sim/FairMCApplication.h (hunk 1 above)
Comment (ChristianTackeGSI) [reply base/sim/FairMCApplication.h:306]: I think, I only changed `fRadGridMan`.
This one was and still is marked with `//!`.

From looking at [Chappter 13.3.2 Transient Data Members](https://root.cern/root/htmldoc/guides/users-guide/InputOutput.html#transient-data-members) it seems that this marks a member variable as not taking part in marshalling.

I had assumed, that the version marking is resposible for on-disk versions?

Comment (fuhlig1) [reply base/sim/FairMCApplication.h:306]: You are right. Normally I increase the version when I change something but in this case it shouldn't be necessary.

**Comment by ChristianTackeGSI** [comment]
So, the change request has been resolved.

What's missing so that this can be merged?

**Review by fuhlig1 — APPROVED** [review]

**Pull request merged**